infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/project/pvc/PVCSubPathHelper.java

```diff
@@ -0,0 +1,240 @@
+import static org.eclipse.che.workspace.infrastructure.openshift.project.OpenShiftObjectUtil.newVolumeMount;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Predicate;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import io.fabric8.kubernetes.api.model.Container;
+import io.fabric8.kubernetes.api.model.ContainerBuilder;
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.api.model.PodBuilder;
+import io.fabric8.kubernetes.api.model.Quantity;
+import java.util.Arrays;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.stream.Stream;
+import javax.annotation.PreDestroy;
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import org.eclipse.che.api.workspace.server.spi.InfrastructureException;
+import org.eclipse.che.commons.lang.concurrent.LoggingUncaughtExceptionHandler;
+import org.eclipse.che.commons.lang.concurrent.ThreadLocalPropagateContext;
+import org.eclipse.che.workspace.infrastructure.openshift.project.OpenShiftPods;
+import org.eclipse.che.workspace.infrastructure.openshift.project.OpenShiftProjectFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Helps to execute commands needed for workspace PVC preparation and cleanup.
+ *
+ * <p>Creates a short-lived Pod based on CentOS image which mounts a specified PVC and executes a
+ * command (either {@code mkdir -p <path>} or {@code rm -rf <path>}). Reports back whether the pod
+ * succeeded or failed. Supports multiple paths for one command.
+ *
+ * <p>Note that the commands execution is needed only for {@link CommonPVCStrategy}.
+ *
+ * @author amisevsk
+ * @author Anton Korneta
+ */
+@Singleton
+public class PVCSubPathHelper {
+
+  private static final Logger LOG = LoggerFactory.getLogger(PVCSubPathHelper.class);
+  private static final JobFinishedPredicate POD_PREDICATE = new JobFinishedPredicate();
+
+  static final int COUNT_THREADS = 4;
+  static final int WAIT_POD_TIMEOUT_MIN = 5;
+
+  static final String[] RM_COMMAND_BASE = new String[] {"rm", "-rf"};
+  static final String[] MKDIR_COMMAND_BASE = new String[] {"mkdir", "-p"};
+
+  static final String IMAGE_PULL_POLICY = "IfNotPresent";
+  static final String POD_RESTART_POLICY = "Never";
+  static final String POD_PHASE_SUCCEEDED = "Succeeded";
+  static final String POD_PHASE_FAILED = "Failed";
+  static final String JOB_MOUNT_PATH = "/tmp/job_mount";
+
+  private final String pvcName;
+  private final String jobImage;
+  private final String jobMemoryLimit;
+  private final OpenShiftProjectFactory factory;
+  private final ExecutorService executor;
+
+  @Inject
+  PVCSubPathHelper(
+      @Named("che.infra.openshift.pvc.name") String pvcName,
+      @Named("che.openshift.jobs.memorylimit") String jobMemoryLimit,
+      @Named("che.openshift.jobs.image") String jobImage,
```

Comment:
Let's rename these properties to something like `che.infra.openshift.pvc.jobs....`

Reply:
Also don't forget to add default values to `che.properties`